rllib/examples/rl_modules/classes/autoregressive_actions_rlm.py

```diff
@@ -0,0 +1,295 @@
+        prior_action = torch.argmax(
+            prior_action_dist._flat_child_distributions[0].rsample(),
+            dim=-1,
+        )
+        # We need the log probability of the sampled action for the loss calculation.
+        prior_action_logp = prior_action_dist._flat_child_distributions[0].logp(
+            prior_action
+        )
+
+        # Posterior forward pass.
+        posterior_batch = torch.cat([batch, prior_action.view(-1, 1)], dim=-1)
+        posterior_out = self.posterior(posterior_batch)
+        # Concatenate the prior and posterior logits to get the final logits.
+        posterior_logits = torch.cat([prior_out, posterior_out], dim=-1)
+        # Get the posterior action distribution to sample the posterior action.
+        posterior_action_dist = self.action_dist_cls.from_logits(posterior_logits)
+        posterior_action = posterior_action_dist._flat_child_distributions[1].sample()
+        posterior_action_logp = posterior_action_dist._flat_child_distributions[1].logp(
+            posterior_action
+        )
+
+        # Concatenate the prior and posterior actions and log probabilities.
+        pi_outs[Columns.ACTIONS] = (prior_action, posterior_action)
+        pi_outs[Columns.ACTION_LOGP] = torch.cat(
+            [prior_action_logp, posterior_action_logp], dim=-1
+        )
+        pi_outs[Columns.ACTION_DIST_INPUTS] = posterior_logits
+        return pi_outs
+
+    @override(TorchRLModule)
+    def _forward_inference(self, batch: Dict[str, TensorType]) -> Dict[str, TensorType]:
+
+        # Encoder forward pass.
+        encoder_out = self.encoder(batch)
+
+        # Policy head forward pass.
+        return self.pi(encoder_out[ENCODER_OUT])
+
+    @override(TorchRLModule)
+    def _forward_exploration(
+        self, batch: Dict[str, TensorType], **kwargs
+    ) -> Dict[str, TensorType]:
+        return self._forward_inference(batch)
+
+    @override(TorchRLModule)
+    def _forward_train(self, batch: Dict[str, TensorType]) -> Dict[str, TensorType]:
+
+        outs = {}
+
+        # Encoder forward pass.
+        encoder_out = self.encoder(batch)
+
+        # Policy head forward pass.
+        outs.update(self.pi(encoder_out[ENCODER_OUT]))
+
+        # Value function head forward pass.
+        vf_out = self.vf(encoder_out[ENCODER_OUT])
+        outs[Columns.VF_PREDS] = vf_out.squeeze(-1)
+
+        return outs
+
+    @override(AutoregressiveActionRLM)
+    def _compute_values(self, batch, device=None):
+        infos = batch.pop(Columns.INFOS, None)
+        batch = convert_to_torch_tensor(batch, device=device)
+        if infos is not None:
+            batch[Columns.INFOS] = infos
+
+        # Encoder forward pass.
+        encoder_outs = self.encoder(batch)[ENCODER_OUT]
+
+        # Value head forward pass.
+        vf_out = self.vf(encoder_outs)
+
+        # Squeeze out last dimension (single node value head).
+        return vf_out.squeeze(-1)
+
+
+# from gymnasium.envs.registration import register
```

Comment:
TODO:
* Move this to separate example script
* (or into existing one) and rewrite in the new API examples style
** self-executable (use `test_utils.run_rllib_example_...` `add_add_rllib_example_script_args`)
** docstring explaining what it is, what it does, what it teaches us, how to run it, what types of envs to run it on, what results to expect

Reply:
Yes that was one of my questions: should we separate definition and usage into two files. Thanks for your opinion on this. 
This is still an initial draft that should get cleaned up in the next iteration.